service-permissions/src/main/java/org/oskari/permissions/PermissionServiceMybatisImpl.java

```diff
@@ -0,0 +1,179 @@
+package org.oskari.permissions;
+
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.db.DatasourceHelper;
+import fi.nls.oskari.domain.Role;
+import fi.nls.oskari.domain.User;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.mybatis.MyBatisHelper;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.oskari.permissions.model.Permission;
+import org.oskari.permissions.model.PermissionExternalType;
+import org.oskari.permissions.model.Resource;
+import org.oskari.permissions.model.ResourceType;
+
+import javax.sql.DataSource;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Oskari
+public class PermissionServiceMybatisImpl extends PermissionService {
+
+    private static final Class<ResourceMapper> MAPPER = ResourceMapper.class;
+    private static final Logger LOG = LogFactory.getLogger(PermissionServiceMybatisImpl.class);
+
+    private final SqlSessionFactory factory;
+
+    public PermissionServiceMybatisImpl() {
+        this(DatasourceHelper.getInstance().getDataSource());
+    }
+
+    public PermissionServiceMybatisImpl(DataSource ds) {
+        if (ds == null) {
+            LOG.warn("DataSource was null, all future calls will throw NPEs!");
+            factory = null;
+        } else {
+            factory = MyBatisHelper.initMyBatis(ds, MAPPER);
+        }
+    }
+
+
+    public List<Resource> findResourcesByUser(User user) {
+        try (SqlSession session = factory.openSession()) {
+            List<Resource> all = session.getMapper(MAPPER).findAll();
+            /*
```

Comment:
WIP? Nothing user specific here

Reply:
Sure, I thought it better to have the API closer to what is needed but the query hasn't been optimized/result filtered yet.

Reply:
Actually I think having resourcetype as second param would benefit this as we can filter out unneeded resources.

Reply:
Sounds like a good idea